src/frontend/src/optimizer/plan_node/logical_scan.rs

```diff
@@ -581,12 +581,33 @@ impl LogicalScan {
             return None;
         }
 
+        let output_col_map = self
+            .output_col_idx()
+            .iter()
+            .cloned()
+            .enumerate()
+            .map(|(id, col)| (col, id))
+            .collect::<BTreeMap<_, _>>();
+        let unmatched_idx = output_col_map.len();
         let index = self.indexes().iter().find(|idx| {
+            let s2p_mapping = idx.secondary_to_primary_mapping();
```

Comment:
Optional: I think `secondary_to_primary_mapping` is not very clear here. Maybe `index_col_to_table_col_mapping`? I prefer something slightly more verbose but clearer.

(Out of scope for this PR, since the method was already written before it. May ignore and leave for future change.)